lib/galaxy/datatypes/binary.py

```
@@ -3357,7 +3357,8 @@ def set_meta(self, dataset, overwrite=True, **kwd):
             if dataset and tarfile.is_tarfile(dataset.file_name):
                 with tarfile.open(dataset.file_name, "r") as temptar:
                     pg_version_file = temptar.extractfile("postgresql/db/PG_VERSION")
-                    dataset.metadata.version = util.unicodify(pg_version_file.read()).strip()
+                    if pg_version_file:
```

maybe raise an exception if there's no pg_version_file ?

---

Done.